src/frontend/src/instance.rs

```diff
@@ -571,17 +573,26 @@ impl PromHandler for Instance {
         query: &PromQuery,
         query_ctx: QueryContextRef,
     ) -> server_error::Result<Output> {
+        let interceptor_ref = self
+            .plugins
+            .get::<PromQueryInterceptorRef<server_error::Error>>();
+        let interceptor = interceptor_ref.as_ref();
```

Comment:
Perhaps we can remove this `as_ref` if the auto trait implementation is based on `Option<PromQueryInterceptorRef<E>>`?